nixos/modules/virtualisation/vmware-guest.nix

```diff
@@ -59,6 +59,26 @@ in
 
     environment.systemPackages = [ cfg.package ];
 
+    security.pam.services.vmtoolsd = {
+      text = ''
+        # Account management.
+        account required ${pkgs.pam}/lib/security/pam_unix.so
```

Comment:
In looking at other usages of `security.pam.services` in the tree we shouldn't have to pass `pkgs.pam` around. I think we might be missing something but I'm not familiar with pam.

Reply:
You can check on your side all files in /etc/pam.d have /nix/store/

For example passwd : 

```
cat /etc/pam.d/passwd
# Account management.
account required /nix/store/fgdhjn2n5gj6gb4szz20kvkibg9qih1b-linux-pam-1.6.1/lib/security/pam_unix.so # unix (order 10900)

# Authentication management.
auth sufficient /nix/store/fgdhjn2n5gj6gb4szz20kvkibg9qih1b-linux-pam-1.6.1/lib/security/pam_unix.so likeauth try_first_pass # unix (order 11600)
auth required /nix/store/fgdhjn2n5gj6gb4szz20kvkibg9qih1b-linux-pam-1.6.1/lib/security/pam_deny.so # deny (order 12400)

# Password management.
password sufficient /nix/store/fgdhjn2n5gj6gb4szz20kvkibg9qih1b-linux-pam-1.6.1/lib/security/pam_unix.so nullok yescrypt # unix (order 10200)

# Session management.
session required /nix/store/fgdhjn2n5gj6gb4szz20kvkibg9qih1b-linux-pam-1.6.1/lib/security/pam_env.so conffile=/etc/pam/environment readenv=0 # env (order 10100)
session required /nix/store/fgdhjn2n5gj6gb4szz20kvkibg9qih1b-linux-pam-1.6.1/lib/security/pam_unix.so # unix (order 10200)
session required /nix/store/fgdhjn2n5gj6gb4szz20kvkibg9qih1b-linux-pam-1.6.1/lib/security/pam_limits.so conf=/nix/store/2sj4krpwpmbff9qbxg36pzj3n1v166sq-limits.conf # limits (order 12200)
```

Reply:
We definitely should not use pkgs.pam here but config.security.pam.package like https://github.com/NixOS/nixpkgs/blob/e6914ee14cc9ef2221e9ac323c59cab1ae029684/nixos/modules/security/pam.nix#L672